cpp/src/arrow/flight/test-util.h

```diff
@@ -0,0 +1,147 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <boost/process.hpp>
+
+#include "arrow/ipc/test-common.h"
+#include "arrow/status.h"
+#include "arrow/test-util.h"
+
+#include "arrow/flight/api.h"
+#include "arrow/flight/internal.h"
+
+namespace bp = boost::process;
+
+namespace arrow {
+namespace flight {
+
+// ----------------------------------------------------------------------
+// Fixture to use for running test servers
+
+struct TestServer {
+ public:
+  explicit TestServer(int port) : port_(port) {}
+
+  void Start() {
+    std::string str_port = std::to_string(port_);
+    server_process_.reset(
+        new bp::child(bp::search_path("flight-test-server"), "-port", str_port));
```

Comment:
I wasn't really sure how to "discover" the test server, even if it's in the same directory as the test executable